src/BackupChain.cs

```diff
@@ -19,26 +20,33 @@ public class BackupChain : IBackupChain
   {
     private readonly IList<BackupMetadata> _orderedBackups;
 
+    /// <summary>
+    /// Chain needs to be 1 full backup, 1 diff and then the logs in the order they were taken
+    /// Sometimes backups can be striped (i.e. split into multiple files) - so we need to handle these cases too
+    /// https://www.sqlservercentral.com/articles/getting-a-list-of-the-striped-backup-files
+    /// </summary>
+    /// <param name="recentBackups"></param>
     private BackupChain(IList<BackupMetadata> recentBackups)
     {
       var backups = recentBackups.Distinct(new BackupMetadataEqualityComparer())
-        .Where(b => IsValidFilePath(b)) // A third party application caused invalid path strings to be inserted into backupmediafamily
+        .Where(IsValidFilePath) // A third party application caused invalid path strings to be inserted into backupmediafamily
         .ToList();
 
-      var mostRecentFullBackup = MostRecentFullBackup(recentBackups);
-      _orderedBackups = new List<BackupMetadata> { mostRecentFullBackup };
-
-      var differentialBackup = MostRecentDifferentialBackup(backups, mostRecentFullBackup);
-      if(differentialBackup != null)
-        _orderedBackups.Add(differentialBackup);
-
-      var mostRecentBackup = _orderedBackups.Last();
-      while(mostRecentBackup != null) {
-        mostRecentBackup = NextLogBackup(backups, mostRecentBackup);
-
-        if(mostRecentBackup != null)
-          _orderedBackups.Add(mostRecentBackup);
+      var mostRecentFullBackups = MostRecentFullBackups(recentBackups).ToList();
+      
+      var differentialBackups = MostRecentDifferentialBackups(backups, mostRecentFullBackups.First());
+      // differentialBackups can be null
+      
+      var orderedBackups = mostRecentFullBackups.Concat(differentialBackups).ToList();
+
+      var prevBackup = orderedBackups.Last();
+      IEnumerable<BackupMetadata> nextLogBackups;
+      while((nextLogBackups = NextLogBackups(backups, prevBackup)).Any()) {
+        orderedBackups.AddRange(nextLogBackups);
+        prevBackup = orderedBackups.Last();
       }
+
+      _orderedBackups = orderedBackups;

```

Comment:
I wonder if it's not easier to just get all of the backup files matching the `LastFullBackup.DatabaseBackupLsn`, then ordering all of them on LastLsn/name at the end

Reply:
Just to clarify, you mean get the **`LastFullBackup.CheckpointLsn`** and then match all backups that have that same value in `backup.DatabaseBackupLsn`?

hmm that could work, but we would need to also filter out any transaction/diff backups that happened before the most recent diff backup - but that should be a simple LINQ query or List remove

I'll have a look at doing that

Reply:
ahhh yeah that's why we chain.... nevermind don't do that